promql/parser/functions.go

```diff
@@ -382,6 +382,24 @@ var Functions = map[string]*Function{
 		Variadic:   1,
 		ReturnType: ValueTypeVector,
 	},
+	"ydelta": {
+		Name:       "ydelta",
+		ArgTypes:   []ValueType{ValueTypeMatrix},
+		ReturnType: ValueTypeVector,
+		ExtRange:   true,
+	},
+	"yincrease": {
+		Name:       "yincrease",
+		ArgTypes:   []ValueType{ValueTypeMatrix},
+		ReturnType: ValueTypeVector,
+		ExtRange:   true,
+	},
+	"yrate": {
+		Name:       "yrate",
+		ArgTypes:   []ValueType{ValueTypeMatrix},
+		ReturnType: ValueTypeVector,
+		ExtRange:   true,
```

Comment:
Can you remind me what `ExtRange: true` is doing here for us?

Reply:
Yes, that's used [over here](https://github.com/Invoca/prometheus/blob/invoca-2.35.0/add-yrate/promql/engine.go#L817-L819) to decide if we need to get `lookbackDelta` (5 minutes by default) of extra samples before the range.

BTW in the Thanos copy of all this code, they skipped that clean table-driven approach and instead just made [a function](https://github.com/thanos-community/promql-engine/blob/21740b7b9fc60089b03e0979137394ca2254ee51/execution/function/functions.go#L938-L940) that knows when they need an "extended range".